### PR TITLE
Add drawer class

### PR DESCRIPTION
**Pull request by davidpmccormick**

### CODE REVIEW

File: client/js/components/drawer.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+class Drawer {
+  constructor(el) {
+    this.el = el;
+    this.trigger = el.getElementsByClassName('js-trigger')[0];
+    this.trigger.addEventListener('click', this.toggleActive.bind(this));
```
Comment (davidpmccormick) [inline client/js/components/drawer.js]: I don't know what the state of cross-browser event handling is in the absence of a library to fall back on – how best to handle this?

File: client/js/components/drawer.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+class Drawer {
+  constructor(el) {
+    this.el = el;
+    this.trigger = el.getElementsByClassName('js-trigger')[0];
```
Comment (jamesgorrie) [inline client/js/components/drawer.js]: What happens if this is an empty array? We would get errors here.

File: client/js/components/drawer.js (hunk 1 above)
Comment (jamesgorrie) [inline client/js/components/drawer.js]: As this is a class we shouldn't have to be binding here, or if we have to, we're probably using it's scope wrong.

Comment (jamesgorrie) [inline client/js/components/drawer.js]: `addEventListenter` [is perfectly fine though](http://caniuse.com/#feat=addeventlistener). We will start having to decide what we consider a % rate where we don't support the browser.

@gestchild?

File: client/js/components/drawer.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    this.trigger.addEventListener('click', this.toggleActive.bind(this));
+  }
+
+  getActive() {
+    return this.el.classList.contains('is-active');
+  }
+
+  setActive(value) {
+    if (value) {
+      this.el.classList.add('is-active');
```
Comment (jamesgorrie) [inline client/js/components/drawer.js]: Tying DOM and state isn't a very long term solution... let's have a think.

File: client/js/components/drawer.js (hunk 1 above)
Comment (gestchild) [inline client/js/components/drawer.js]: The list I came up with covers over 95% of traffic. See below:

Chrome (latest 2 versions)	Full support
Safari (latest 2 versions)	Full support
Firefox (latest 2 versions)	Full support
Edge (latest 2 versions)	        Full support
Internet Explorer (latest 2 versions)	Full support
Android Browser	Full support

Opera Mini (extreme mode)	Basic support 

And providing we're using 'cutting the mustard' and progressive enhancement, anything not supporting addEventListener should get the non javascript version

Comment (jamesgorrie) [inline client/js/components/drawer.js]: Yea - IE8 being one that won't, and I would heavily suggest for our sanity that it gets nowhere near the mustard.

(but it still get's the core content).

File: client/js/components/drawer.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+      this.el.classList.remove('is-active');
+    }
+  }
+
+  toggleActive() {
+    this.setActive(!this.getActive());
+  }
+}
+
+export default Drawer;
```
Comment (jamesgorrie) [inline client/js/components/drawer.js]: 👍  to finding a simpler solution.
Probably better too - as you'd want to see all of them, unless the list get's unwieldy - which maybe means we should start wondering why we have so many components.

Comment (jamesgorrie) [inline client/js/components/drawer.js]: It brings up a good conversation though.

Comment (davidpmccormick) [inline client/js/components/drawer.js]: At some point we're going to have to decide at what level of granularity we're going to 'componentise' things, and then I think we'll also have to decide whether we want each and every _thing_ that has a JS component aspect to it to be displayed in the pattern library (I would think probably not – perhaps we'd just want to display larger, complete, composed things).

Meantime, I'll remove this component and display everything in the sidebar.

Comment (jamesgorrie) [inline client/js/components/drawer.js]: Yip - and probably start using the new language (ingredients, atoms, quarks).

**Comment by davidpmccormick** [comment]
Yeah – would be nice. Probably not mandatory – let pragmatism prevail?

**Pull request merged**